tests/compile_time/compile_time_json_tests.cpp

```diff
@@ -498,7 +616,11 @@ bool run() {
            test_simple_object_str_with_obj() &&
            test_empty_arrays() &&
            simple_array() &&
-           array_of_objects();
+           test_complex_mixed_str() &&
+           array_of_objects() &&
+           test_user_config_example() &&
+           test_nested_servers_example() &&
+           test_top_level_array_example();
```

Comment:
@lemire do you think we are missing a case where the json comes from an external file known at compile-time? (embed can be used for that)

Other than that, this is pretty awesome to see. Thank you for your work (: